azure/sdk-identity-resources-storage/README.md

```diff
@@ -0,0 +1,71 @@
+﻿## Prerequisites
+
+* An [Azure subscription]().
+
+## Preparing to run the sample
+
+Use the [Azure Cloud Shell](https://shell.azure.com) to create/get client secret credentials.
+
+* Create a service principal and configure its access to Azure resources:
+
+    ```bash
+    az ad sp create-for-rbac -n <your-application-name> --skip-assignment
+    ```
+
+    Output:
+
+    ```json
+    {
+        "appId": "generated-app-ID",
+        "displayName": "dummy-app-name",
+        "name": "http://dummy-app-name",
+        "password": "random-password",
+        "tenant": "tenant-ID"
+    }
+    ```
+
```

Comment:
You'll also probably want to add some of https://docs.microsoft.com/en-us/azure/storage/common/storage-auth-aad-rbac-cli to assign roles to your service principal.

Reply:
@tg-msft This is actually what was holding me back in regards to using `DefaultAzureCredential`. Do you have any guidance on how to do the RBAC assignments programmatically? I'd prefer to not have the user use AzCLI to do that and I'd prefer to show them in code.

Reply:
I'm not as well informed about [the management plane APIs](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-rest), but [a search for `Microsoft.Authorization/roleAssignments` in the azure-sdk-for-net repo](https://github.com/Azure/azure-sdk-for-net/search?l=C%23&p=2&q=Microsoft.Authorization%2FroleAssignments) makes me think the Azure.ResourceManager.Resources folks haven't added that explicitly yet.  It looks like it [might be possible](https://docs.microsoft.com/en-us/dotnet/api/azure.resourcemanager.resources.models.applicationproviderauthorization?view=azure-dotnet-preview) as [part of](https://docs.microsoft.com/en-us/dotnet/api/azure.resourcemanager.resources.models.applicationdefinition?view=azure-dotnet-preview) creating an [application definition](https://docs.microsoft.com/en-us/dotnet/api/azure.resourcemanager.resources.applicationdefinitionsoperations.startcreateorupdateasync?view=azure-dotnet-preview#Azure_ResourceManager_Resources_ApplicationDefinitionsOperations_StartCreateOrUpdateAsync_System_String_System_String_Azure_ResourceManager_Resources_Models_ApplicationDefinition_System_Threading_CancellationToken_), but I have no idea what that entails.  I'd set it up with CLI for the time being and then update the sample in the future when it's possible to do so in code.

//cc @markcowl who can hopefully correct me if I'm wildly off base